### ISSUE TITLE
[BUG] can't install along with tox

### ISSUE THREAD

**Issue by medecau** [PyCQA/prospector#579]
Dependency version resolution is failing when installing in a project with tox.

Poetry's output with all the relevant details:
```
Because no versions of prospector match >1.8.4,<2.0.0
 and prospector (1.8.4) depends on packaging (<22.0), prospector (>=1.8.4,<2.0.0) requires packaging (<22.0).
And because tox (4.2.8) depends on packaging (>=23)
 and no versions of tox match >4.2.8,<5.0.0, prospector (>=1.8.4,<2.0.0) is incompatible with tox (>=4.2.8,<5.0.0).
So, because pyreviewer depends on both tox (^4.2.8) and prospector (^1.8.4), version solving failed.
```
The issue seems arise from a `prospector` dependency on `packaging<22.0`.

**To Reproduce** Steps to reproduce the behavior:

1. `take` a new directory
2. `poetry init`
3. `poetry add tox`
4. `poetry add prospector`
5. Observe error

- OS: macOS
- Tool prospector
- Prospector version ^1.8
- Python version 3.11
- Poetry version 1.3.2

Note that I have successfully added prospector as a dependency with `prospector<1.8`.

**Comment by koniiiik** [comment]
Is there a reason why the packaging dependency is pinned the way it is? Briefly looking through the changelog of packaging, the API seems pretty stable, and the backwards incompatibilities mostly amount to dropping support for old Python, or removing obscure bits of API that prospector isn't making use of anyway. Moreover, the versioning scheme seems to be mostly based on the year the release was made, rather than semantic versioning.

The version restriction itself was added in https://github.com/PyCQA/prospector/pull/576/commits/cea00c37b1902f95b425daba9b6730b60f0715a7 without any discussion…

**Comment by Pierre-Sassoulas** [comment]
See #576 for rational, I'm open to new arguments of course :)

**Comment by koniiiik** [comment]
I'm not challenging having packaging listed as a dependency – that makes perfect sense, since it is a package whose API is directly being used by prospector. The thing I'm challenging is restricting its version to `<22.0`. Is there any argument against allowing newer versions? Why was it pinned not even to the latest version at the time the dependency was introduced?

**Comment by Pierre-Sassoulas** [comment]
Right, let's release the constraint. Do you want to open a merge request ?

**Comment by ryancausey** [comment]
Looks like this won't work unless `requirements-detector` widens or bumps their range also:

```bash
$ poetry add "packaging=^22.0"

Updating dependencies
Resolving dependencies... (0.7s)

Because requirements-detector (1.0.3) depends on packaging (>=21.3,<22.0)
 and no versions of requirements-detector match >1.0.3, requirements-detector (>=1.0.3) requires packaging (>=21.3,<22.0).
So, because prospector depends on both requirements-detector (>=1.0.3) and packaging (^22.0), version solving failed.
```

This is because black 23.1.0 depends on `packaging >22.0`

```bash
Because black (23.1.0) depends on packaging (>=22.0)
 and prospector (1.8.4) depends on packaging (<22.0), black (23.1.0) is incompatible with prospector (1.8.4).
So, because test-package depends on both black (23.1.0) and prospector (1.8.4), version solving failed.
```

**Comment by Pierre-Sassoulas** [comment]
@carlio do you think it's reasonable to replace ``requirements-detector`` by some custom code ?

**Comment by ryancausey** [comment]
I opened a PR against `requirements-detector` to bump the version range, but it also appears to require dropping python 3.6 support.

**Comment by ryancausey** [comment]
The PR against requirements detector was merged and released in requirements-detector 1.1.0. It looks like this issue might be resolved by #592 however?

**Comment by Pierre-Sassoulas** [comment]
#592 was released in prospector 1.9.0

**Comment by medecau** [comment]
Just checked with tox 4.4.6 and prospector 1.9.0, and it all works as expected.

Thank you all for working to fix this. 🙏🏼
